climada/util/coordinates.py

```diff
@@ -1821,6 +1821,160 @@ def apply_box(df_exp):
     }
     return raster_out, meta
 
+def reproject_raster_data(source, src_crs, src_transform, dst_crs=None, dst_transform=None,
+                      dst_shape=None, dst_resolution=None, dst_global_origin=None,
+                      dst_global_extent=None,
+                      dst_buffer=0, resampling=None, conserve=None,
+                      **kwargs):
+    """
+    Convenience wrapper function around rasterio.warp.reproject :
+    Reproject (up- or downsample) 2D np.ndarray to a grid using rasterio.
+    The destination (target) grid is defined from dst_transform and optionally
+    dst_resolution, dst_global_origin, and dst_crs.
+    This function ensures that reprojected data with the same dst_resolution and
+    dst_global_origins are on the same global grid, i.e., no offset between
+    destination grid points for different source grids that are projected to
+    the same target resolution.
+
+    Parameters
+    ----------
+    source : np.ndarray
+        2D-array containing the values to be reprojected
+    src_crs : CRS
+        Source CRS
+    src_transform : Affine
+        Source transform, contains info on grid, e.g.:
+            Affine(0.00833333333333333, 0.0, -18.175000000000068,
+                                 0.0, -0.00833333333333333, 43.79999999999993)
+    dst_crs : CRS, optional
+        Destination CRS. The default is src_crs.
+    dst_transform : Affine, optional
+        Destination transform. The default is src_transform.
+    dst_shape : tuple with integers, optional
+        (height, width) of destination grid. The default is source.shape.
+        Note: Destination shape is changed if resolution changes
+    dst_resolution : number, optional
+        Destination resolution.
+        Overwrites resolution provided in dst_transform.
```

Comment:
I suggest not to overwrite one user-specified parameter by another, but to define certain combinations of input parameters as illegal and raise an exception. The most obvious example would be `dst_transform` together with `dst_resolution` (or `dst_global_origin`).

Reply:
I guess there is a misunderstanding here, the whole idea is to reproject to a consistent global grid based both on dst_transform and dst_resolution. However, I will avoid future confusions by not overwriting dst_transform and rather defining an new dst_transform_newres if provided, will also adjust docstring.

Reply:
Hm, I'm not sure what you mean. Why would a user provide the `dst_transform` at all, if they want the information therein to be overwritten by other parameters?

Reply:
The idea of this function is to reproject either to a new resolution based on the source grid (src_transform and dst_resolution are required input to define the destination grid), reproject to a given grid (dst_transform defines the destinatiion grid, no dst_resolution given), or reproject to a new resolution on a grid that is globally consistent to the grid defined in dst_transform, in this case both dst_transform and dst_resolution need to be provided. This later application is exactly what I need for LitPop, i.e. project Lit-data (source) on a grid with given resolution dst_resolution that is defined consistently with the Pop-grid (defined in dst_transform). Providing both dst_transform and dst_resolution makes sure there is no offset between coordinates of Lit and Pop both reprojected to the same resolution X.

I will add an according explanation to the doc string.

Reply:
i will also rename dst_transform to reference_transform to avoid confusion

Reply:
I still don't understand what you mean. How can a grid be "globally consistent" with a reference transformation if the resolution is overwritten by another value? What do you mean by "globally consistent"?

Reply:
Given an equidistant 2D-grid defined by src_transform, crs and (optionally)
        src_shape, a destination grid with a new resolution (dst_resolution) is defined.
        The destination grid is globally consistent relative to a global origin.
        I.e., there is no offset in coordinates compared to another grid reprojected to
        the same dst_resolution based on a different src_transform.